indigo_app/forms.py

```diff
@@ -305,7 +305,10 @@ def filter_queryset(self, queryset):
         
         # filter by subtype indicated on frbr_uri
         if self.cleaned_data.get('subtype'):
-            queryset = queryset.filter(frbr_uri__contains='/act/%s/' % self.cleaned_data['subtype'].abbreviation)
+            if self.cleaned_data['subtype'] == 'act':
+                queryset = queryset.filter(frbr_uri__regex=r'.\/act\/\d{4}\/\w+')
```

Comment:
Nice. At some point we should add subtype as a field on the model so we don't have to do this weirdness.